core/src/main/java/org/elasticsearch/cluster/node/DiscoveryNode.java

```diff
@@ -338,25 +336,24 @@ public Version getVersion() {
     }
 
     public static DiscoveryNode readNode(StreamInput in) throws IOException {
-        DiscoveryNode node = new DiscoveryNode();
-        node.readFrom(in);
-        return node;
+        return PROTOTYPE.readFrom(in);
```

Comment:
can we get rid of `PROTOTYPE ` and remove the readNode method alltogether. I thikn we should just have a `DiscoveryNode(StreamInput in)` method and do all the reading there.